logic/transfer.js

```diff
@@ -166,32 +197,102 @@ Transfer.prototype.undoUnconfirmed = function (trs, sender, cb) {
 	return setImmediate(cb);
 };
 
+
+/**
+ * @typedef {Object} transfer 
+ * @property {String} data
+ */
+Transfer.prototype.schema = {
+	id: 'transefer',
+	type: 'object',
+	properties: {
+		data: {
+			type: 'string',
+			minLength: 0,
+			maxLength: 64
+		}
+	}
+};
+
 /**
- * Deletes blockId from transaction 
+ * Deletes blockId from transaction, and validates schema if asset exists
  * @param {transaction} trs
  * @return {transaction}
  */
 Transfer.prototype.objectNormalize = function (trs) {
 	delete trs.blockId;
+
+	if (!trs.asset) {
+		return trs;
+	}
+
+	for (var i in trs.asset) {
+		if (trs.asset[i] === null || typeof trs.asset[i] === 'undefined') {
```

Comment:
Asset can be only one. But the important thing to note is that `data` field is attached like this:
```
{
  asset: {
    data: '...'
  }
}
```
And it's not like this:
{
  asset: {
    transfer: {
        data: '...'
   }
  }
}